### ISSUE TITLE
Visiting a Route with param, navigate function or Link doesn't reset the param when navigating to another route

### ISSUE THREAD

**Issue by TommysG** [TanStack/router#1140]
### Describe the bug

I have a problem when visiting directly a route with param. For example: [http://localhost:3000/posts/4](http://localhost:3000/posts/4 ).

After visiting the route, when I tap a button with the `navigate` function, it doesn't clear the previous param.

```ts
<button onClick={() => navigate({to: '/posts'})}>Close Modal</button>
```

### Your Example Website or App

https://stackblitz.com/~/edit/github-844nb4?file=src%2Fmain.tsx

### Steps to Reproduce the Bug or Issue

1. go to `/posts/$postId` directly.
2. tap a button that navigates you back to `/posts`
3. the param doesn't reset

### Expected behavior

As a user i expect the param to be cleared

### Platform

- OS:macOS
- Browser:  Chrome
- @tanstack/react-router: 1.15.13

**Comment by schiller-manuel** [comment]
I was able to reproduce this:
https://github.com/TanStack/router/assets/6340397/da177b71-8ca3-4cb0-9fd7-885ac8bf9647

even though the route does not have path params, they are still in the router state (see devtools in screen recording)

**Comment by schiller-manuel** [comment]
fixed by cde563d5e9c1a79b14c652171f2b31ad7cd6929d
